Week3/homework/step2-2.js

```diff
@@ -2,21 +2,25 @@
 
 function threeFive(startIndex, stopIndex, threeCallback, fiveCallback) {
   const numbers = [];
-
-  // Replace this comment and the next line with your code
-  console.log(startIndex, stopIndex, threeCallback, fiveCallback, numbers);
+  for (let i = startIndex; i <= stopIndex; i++) {
+    numbers.push(i);
+    if (i % 3 === 0 && i % 5 === 0) {
+      threeCallback(i);
+      fiveCallback(i);
+    }
+    if (i % 3 === 0 && i % 5 !== 0) {
+      threeCallback(i);
+    } else if (i % 5 === 0 && i % 3 !== 0) {
+      fiveCallback(i);
+    }
+  }
```

Comment:
I don't think you need to have the if checks as:
```
if (i % 3 === 0 && i % 5 !== 0) { }
```
and
```
else if (i % 5 === 0 && i % 3 !== 0) { }
```

Once you have a check for `i % 3 === 0 && i % 5 === 0` at the beginning, you need not check for `i % 5 !== 0` and `i % 3 !== 0` as it is now redundant. 

Also, a number which is divisible by 3 and 5 both will also be divisible by 15, right? 😉 Can you think of making your first if condition shorter then?

Reply:
Also, I am wondering if you did work together with Faruk on the logic for this particular assignment as we see here - https://github.com/HackYourFuture/JavaScript2/blob/d05c3e0119b49a062edd1609948a22fb957504ab/Week3/homework/step2-2.js? 🙂

Reply:
> I don't think you need to have the if checks as:
> 
> ```
> if (i % 3 === 0 && i % 5 !== 0) { }
> ```
> 
> and
> 
> ```
> else if (i % 5 === 0 && i % 3 !== 0) { }
> ```
> 
> Once you have a check for `i % 3 === 0 && i % 5 === 0` at the beginning, you need not check for `i % 5 !== 0` and `i % 3 !== 0` as it is now redundant.
> 
> Also, a number which is divisible by 3 and 5 both will also be divisible by 15, right? 😉 Can you think of making your first if condition shorter then?

yes that's right, no need the second if statements as the first one does what's needed to be done. when busy with bunch of assignments in a late time, some repetition exists and you miss that. I'll be paying more attention

Reply:
> Also, I am wondering if you did work together with Faruk on the logic for this particular assignment as we see here - https://github.com/HackYourFuture/JavaScript2/blob/d05c3e0119b49a062edd1609948a22fb957504ab/Week3/homework/step2-2.js? 🙂

True. Some time we group up for a part of homework and try to find solution together. And to speak frankly, we just thought yesterday that we better notify instructors when it is the case

Reply:
Got it. Yes, I think it will be quite helpful for the reviewers going through the pull requests. Thanks. 🙂